### PR TITLE
feat: add support for external-secrets

### PR DESCRIPTION
**Pull request by Spikra**
- Adds support for creating externalSecrets resources to sync secrets from secret manager.
- Mounts kubernetes secrets created from the externalSecret resources as env vars.

### CODE REVIEW

File: charts/common/tests/externalsecret_test.yaml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    custom: label
+  ingress:
+    host: test.dev.entur.io
+    trafficType: public
+  secrets:
+    one-secret:
+      - FIRST_SECRET
+    two-secrets:
+      - FIRST_SECRET
+      - SECOND_SECRET
```
Comment (AlexanderBrevig) [inline charts/common/tests/externalsecret_test.yaml:16]:
```suggestion
      - SECOND_SECRET
      - THIRD_SECRET
```

**Review by AlexanderBrevig — CHANGES_REQUESTED** [review]

File: charts/common/tests/externalsecret_test.yaml (hunk 1 above)
Comment (AlexanderBrevig) [reply charts/common/tests/externalsecret_test.yaml:16]: Tests needs to be updated to reflect this disambiguation.

File: charts/common/values.yaml [hunk 2]
```diff
@@ -269,3 +269,10 @@ configmap:
   data: {}
     # MY_CONFIG: yes
 
+secrets:
+# -- add externalSecret to sync secrets from secret manager
+# @default -- {}
```
Comment (AlexanderBrevig) [inline charts/common/values.yaml:274]: Move above `secrets:` stanza so they get included in doc

File: example/values-minimal-secret.yaml
Comment (AlexanderBrevig) [inline example/values-minimal-secret.yaml:1]: do not include this file

File: README.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ Update version in Chart.yaml and run `helm dependency update`
 
 ```bash
 helm template charts/common -f example/values-minimal.yaml
+helm template charts/common -f example/values-minimal-secret.yaml
```
Comment (AlexanderBrevig) [inline README.md:32]: This pattern will quickly hurt (postgres, secrets, sngle container, multi container, custom specs, multi svc,ing etc etc.
I think the absolute minimal for deploy/cron is enough for this repo IMHO.

File: charts/common/tests/externalsecret_test.yaml (hunk 1 above)
Comment (Spikra) [reply charts/common/tests/externalsecret_test.yaml:16]: Renamed the secret in `one-secret` to SINGLE_SECRET, and updated the tests to reflect the change.

File: charts/common/values.yaml (hunk 2 above)
Comment (Spikra) [reply charts/common/values.yaml]: Moved, and removed the default since it is extrapolated from the actual value.

File: README.md (hunk 3 above)
Comment (Spikra) [reply README.md]: I agree. The file was useful when testing the new functionality, but it does not need to be added to the codebase.

**Review by AlexanderBrevig — DISMISSED** [review]

**Review by AlexanderBrevig — APPROVED** [review]

**Pull request merged**